packages/frontend/src/components/QrReader/index.tsx

```diff
@@ -92,7 +94,34 @@ async function base64ToImageData(base64: string): Promise<ImageData> {
   })
 }
 
-export default function QrReader({ onError, onScan }: Props) {
+async function processClipBoard(runtime: Runtime) {
+  // Try interpreting the clipboard data as an image
+  const base64 = await runtime.readClipboardImage()
+  if (base64) {
+    const imageData = await base64ToImageData(base64)
+    const result = scanQrCode(imageData.data, imageData.width, imageData.height)
+    if (result) {
+      return result
```

Comment:
Nitpick: this function is only used in one place, and only the `data` property of this return value is used. I think it'd be simpler to make it return `result.data` only, so there is no need to check `typeof` of the return value if this function.

Reply:
That's true - it is a leftover of the previous handleScanResult which I don't know of why it was neccessary (see my question to simon). There the result was stripped down to result.data

I will reduce that...